crates/rust-analyzer/src/bin/main.rs

```diff
@@ -21,14 +21,25 @@ static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
 fn main() {
     if let Err(err) = try_main() {
+        log::error!("Unexpected error: {}", err);
         eprintln!("{}", err);
         process::exit(101);
     }
 }
 
 fn try_main() -> Result<()> {
     let args = args::Args::parse()?;
-    setup_logging(args.log_file)?;
+
+    #[cfg(debug_assertions)]
+    if args.wait_dbg || env::var("RA_WAIT_DBG").is_ok() {
+        #[allow(unused_mut)]
+        let mut d = 4;
+        while d == 4 {
+            d = 4;
+        }
+    }
```

Comment:
Can you give some kind of indication that flag/env var is ignored when debug assertions are disabled? Or just panic?

Reply:
Ah, I got it. `--wait-dbg` is pretty niche and to my mind it's better not to confuse users. So it is intended that in release builds we just ignore the flag.